### PR TITLE
Auto-Generated Docs!!

### PR DESCRIPTION
**Pull request by samuel-tate**
## Related issue
#13 

## About

I'm using [sphinx autodoc](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html) to automatically generate the documentation using the docstrings.

Some of the documentation looks kinda wack, but overall I think it's pretty nice. LMK what you think though!

## How to view the docs

```
$ cd docs
$ make html
```

There might be a bunch of warnings - I haven't gone through them all yet.

open `./docs/_build/html/index.html` in a browser

### CODE REVIEW

File: .github/workflows/update_docs.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: Update Documentation
+
+on:
```
Comment (kurtwheeler) [inline .github/workflows/update_docs.yml:3]: Is this missing something? Is this the thing that you didn't get working? If that's the case maybe just move it to another branch?

Comment (samuel-tate) [reply .github/workflows/update_docs.yml:3]: Yeah I took this out cause it doesn't work.
I can move to a different branch.

Comment (kurtwheeler) [reply .github/workflows/update_docs.yml]: Thanks!

**Comment by kurtwheeler** [comment]
Can you add something to the readme about how to run the docs locally? Maybe also how to deploy the site to github pages?

File: docs/http.rst [hunk 2]
```diff
@@ -0,0 +1,6 @@
+
+http
+====
+
+.. automodule:: pyrefinebio.http
+   :members:
```
Comment (kurtwheeler) [inline docs/http.rst:6]: This page shows up blank for me.

File: docs/high_level_functions.rst [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
+High Level Functions
+====================
+
+.. automodule:: pyrefinebio.high_level_functions
+   :members:
```
Comment (kurtwheeler) [inline docs/high_level_functions.rst:6]: Is there any way to get some spacing between the functions? This seems cramped and it's hard to scan the different functions to find the one I want:

image

File: docs/sample.rst [hunk 4]
```diff
@@ -0,0 +1,6 @@
+
+Sample
+======
+
+.. autoclass:: pyrefinebio.Sample
+   :members:
```
Comment (kurtwheeler) [inline docs/sample.rst:6]: Are there any levers that Spinx gives us to pull to change the formatting?

image

is a pretty big block that is not easy to read.

**Review by kurtwheeler — APPROVED** [review]
Ok I think this looks pretty good! It might be worth spending some more time improving the styling of these docs, but besides seeing if any of the changes I suggested can be done easily, I think it'd be better to get this out and improve it further in subsequent PRs.

**Review by kurtwheeler — CHANGES_REQUESTED** [review]
Actually now that I think about it the blank HTTP page and the README should definitely be fixed. The styling changes should only be done if they can be done quickly/easily.

**Review by kurtwheeler — APPROVED** [review]
Ok sweet! This looks good enough for now!!

**Pull request merged**